becquerel/core/calibration.py

```diff
@@ -0,0 +1,787 @@
+"""Generic calibration class."""
+
+from abc import abstractmethod
+import ast
+import copy
+import asteval
+import black
+import blib2to3
+import numpy as np
+import scipy
+import scipy.optimize
+from .. import io
+
+CLIP_MAX = 1e6  # maximum value for a calibration function
```

Comment:
This could be improved by implementing domains, I think.

Reply:
Should be resolved now with the addition of `domain` and `range` to `Calibration`.